src/authorizations/components/CustomApiTokenOverlay.tsx

```diff
@@ -106,10 +115,10 @@ const CustomApiTokenOverlay: FC<Props> = props => {
     // BUT, code inside the hook won't run until remoteDataState is 'Done'.
     // Only then will props.bucketPermissions.sublevelPermissions will have value.
     // Consequently, we update the permissions state.
-  }, [props.remoteDataState])
+  }, [remoteDataState]) // eslint-disable-line react-hooks/exhaustive-deps
 
```

Comment:
Same here - what dependencies does the useEffect expect other than `remoteDataState`?  If we can safely include those dependencies, we should likely do so here.

Reply:
aside from `remoteDataState`, it expects `allResources, bucketPermissions, and telegrafPermissions`. Adding the rest will cause the hook to run more times than necessary. Only dependency we're concerned with here is `remoteDataState`. Only when its state changes, we want the code inside the hook to run.

Reply:
Elsewhere, a possible solution is to include an `if` check that will only run the logic when the remote data state is `NotStarted` (or whatever it should be when the logic is run). That way, the logic of the hook runs only when we want it to, but we can also include everything in the dependency array.

Reply:
so what you're suggesting (`if` check containing code that will only run depending on state of remoteDataState) is exactly what the code is currently doing. `if (!isEmpty(bucketPermissions.sublevelPermissions))` will only return true when `remoteDataState` is `Done`. In other words, code inside the if statement won't execute until all necessary data is fetched. But still, including the rest of the values in the dependency array means useEffect will run one more time than necessary. Check out the video: first one is with `remoteDataState` as the only thing in the dependency array. Second video shows console when useEffect has all values in it's array : 

https://user-images.githubusercontent.com/66275100/198349558-38e3a2bd-bf5d-4bc3-b65f-c1f2076ad84c.mov

https://user-images.githubusercontent.com/66275100/198349675-363d090d-1808-4fbb-9a13-19ef5b317636.mov

Reply:
In other words - could you try seeing what happens if you change the `if` check to directly check whether the remotedatastate is `Done` before running the logic (instead of looking for sublevelPermissions?)? If so, I think the logic will only run once, because the interim changes in permissions all happen before the data state is finally set to `Done`

Reply:
yes, we could also change the if statement to directly check `props.remoteDataState === RemoteDataState.Done` but we our dependency array dilemma will still exist. Actually, doing so will cause the useEffect hook to run even more times. check it out: 

https://user-images.githubusercontent.com/66275100/198364241-bb64a9c1-0f84-4cfa-b176-f68787f61a1b.mov

Here is the code: